api/tokens/views.py

```diff
@@ -0,0 +1,98 @@
+"""
+Views related to person access tokens. Intended for OSF internal use only
+"""
+from rest_framework.exceptions import APIException
+from rest_framework import generics
+from rest_framework import renderers
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth import cas
+from framework.auth.oauth_scopes import ComposedScopes
+
+from api.base.filters import ODMFilterMixin
+from api.base.utils import get_object_or_error
+from api.base import permissions as base_permissions
+from api.tokens.serializers import ApiOAuth2PersonalTokenSerializer
+
+from website.models import ApiOAuth2PersonalToken
+
+
+class TokenList(generics.ListCreateAPIView, ODMFilterMixin):
+    """
+    Get a list of personal access tokens that the user has registered
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.OwnerOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [ComposedScopes.FULL_READ]
```

Comment:
Each view should reference a single piece of data- a `CoreScope` referring to this one specific API endpoint / type of view to be serialized. It should never reference `ComposedScopes` at all.

Given that it compares the list of what a token allows to the list in this field... I'm slightly surprised if this works as written.